### PR TITLE
add ci to test template

### PR DESCRIPTION
**Pull request by karthik2804**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -38,4 +38,40 @@ jobs:
...
+
+    steps:
+        - uses: actions/checkout@v3
+
+        - name: Use Node.js 20
+          uses: actions/setup-node@v3
+          with:
+            node-version: 20
+
+
```
Comment (vdice) [inline .github/workflows/test.yaml]: nit: extra newline/whitespace 😅

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -38,4 +38,40 @@ jobs:
...
             cd test
-            ./test.sh
\ No newline at end of file
+            ./test.sh
+  
+  test_template:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (vdice) [inline .github/workflows/test.yaml:43]: Out of curiousity, is there anything os-specific about the test template?  Or are we mostly testing spin/npm on each OS?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -38,4 +38,40 @@ jobs:
...
+        - uses: actions/checkout@v3
+
+        - name: Use Node.js 20
+          uses: actions/setup-node@v3
+          with:
+            node-version: 20
+
+            
+        - name: Install Spin
+          uses: rajatjindal/setup-actions/spin@main
```
Comment (vdice) [inline .github/workflows/test.yaml]: Can we use the official [spin/setup](https://github.com/fermyon/actions?tab=readme-ov-file#install-spin-cli-and-plugins---fermyonactionsspinsetupv1) action instead?  (Nearly all written by rajatjindal 😄)

File: .github/workflows/test.yaml (hunk 2 above)
Comment (karthik2804) [reply .github/workflows/test.yaml:43]: We are just testing to `spin/npm` - particularly the `npm` packages we use in our build step.

File: .github/workflows/test.yaml (hunk 3 above)
Comment (karthik2804) [reply .github/workflows/test.yaml]: Will update.

**Review by vdice — APPROVED** [review]

**Comment by karthik2804** [comment]
Will merge after I fix the failing `windows` test.

**Comment by karthik2804** [comment]
I am thinking just building the app should be sufficient, do we also want to test running the app and getting a response? I do not have string preferences here.

**Comment by vdice** [comment]
> I am thinking just building the app should be sufficient, do we also want to test running the app and getting a response? I do not have string preferences here.

What you've added already seems great for now.  Can be a follow-up if we end up wanting that coverage?

**Pull request merged**